### PR TITLE
List Python3 dependencies

### PR DESCRIPTION
**Pull request by jngrad**
Follow-up to #3093

List the minimal version numbers that are supported for dependent Python modules.

Please note `requirements.txt` does not have to list strict dependencies, optional packages are also allowed. Comments after a package name must have a whitespace before the `#` symbol.

### CODE REVIEW

**Comment by jngrad** [comment]
You can check a list of dependencies against your environment with the following script:
```python
import pip
from pip._internal.req.req_file import parse_requirements
from pip._internal.network.session import PipSession
import pkg_resources
from pkg_resources import DistributionNotFound, VersionConflict

def check_dependencies():
    """
    Checks if python dependencies are fullfilled.
    If check passes, return 0. Otherwise, print error and return 1.
    Please note that after the first missing module is found, pkg_resources
    will ignore all other missing packages!
    Source: https://github.com/pypa/pip/issues/2733#issuecomment-257340871
    """
    requirement_file_name = 'requirements.txt'
    dependencies = []
    session = PipSession()
    for req in parse_requirements(requirement_file_name, session=session):
        if req.req is not None:
            dependencies.append(str(req.req))
        else:
            # The req probably refers to a url. Depending on the
            # format req.link.url may be able to be parsed to find the
            # required package.
            pass
    try:
        pkg_resources.working_set.require(dependencies)
    except VersionConflict as e:
        try:
            print("{} was found on your system, "
                  "but {} is required.\n".format(e.dist, e.req))
            return 1
        except AttributeError:
            print(e)
            return 1
    except DistributionNotFound as e:
        print(e)
        return 1
    print('Check was successful')
    return 0

exit(check_dependencies())
```

**Comment by mkuron** [comment]
With this significantly expanded requirements.txt file, we should remove the recommendation to install it from https://github.com/espressomd/espresso/blob/df08343e108f9d53cfda759c3341d6386adbcc6c/doc/sphinx/contributing.rst#required-development-tools. We already removed it from the user documentation a few weeks ago. Nobody wants to install a dozen packages plus dependencies if they don't actually need them.

File: doc/sphinx/contributing.rst [hunk 1]
```diff
@@ -84,22 +84,26 @@ Required Development Tools
...
--  To build the tutorials, you will need LaTeX and Jupyter.
+You can find all Python dependencies of |es| in :file:`requirements.txt` in the
+top-level source directory. Several optional packages for graphics, external
+devices and continuous integration (CI) are not strictly essential and can be
+safely removed if you're planning on installing dependencies via ``pip``:
 
--  To compile the Doxygen code documentation, you will need the tool doxygen_.
+.. code-block:: bash
+
+   pip install --upgrade --user -r requirements.txt
```
Comment (mkuron) [inline doc/sphinx/contributing.rst]: Change this to `pip3`. That seems to be the default name nowadays.

**Review by mkuron — APPROVED** [review]

File: doc/sphinx/contributing.rst (hunk 1 above)
Comment (jngrad) [reply doc/sphinx/contributing.rst]: Thanks!

Comment (hmenke) [reply doc/sphinx/contributing.rst]: `python3 -m pip` is also a possibility.

**Review by fweik — APPROVED** [review]

**Comment by jngrad** [comment]
bors r=fweik

**Pull request merged**